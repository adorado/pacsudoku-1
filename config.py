class Configfile():
    def __init__(self,inputType="TXT",outputType="Console",defaultAlgorithm="Backtracking",difficultyLevel="Low"):
        """ This is a constructor of a Configfile class."""
        if self._validateInputType(inputType,["TXT","CSV"])==True:
            self.inputType=inputType
        else:
            self.inputType= "TXT"
        
        if self._validateInputType(outputType,["Console","File"])==True:
            self.outputType=outputType
        else:
            self.outputType="Console"
            
        if self._validateInputType(defaultAlgorithm,["Backtracking","Peter Norvig","Other"])==True:    
            self.defaultAlgorithm=defaultAlgorithm
        else:
            self.defaultAlgorithm="Backtracking"
            
        if self._validateInputType(difficultyLevel,["High","Medium","Low"])==True:
            self.difficultyLevel=difficultyLevel
        else:
            self.difficultyLevel="Low"
        
    def _validateInputType(self,inputtype, formats):
        status=False
        for a in range(len(formats)):
            if inputtype==formats[a]:
                status=True
                break 
            else:
                status=False
        return status
