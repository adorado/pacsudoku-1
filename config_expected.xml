<config>    <inputType>CSV</inputType>    <outputType>Console</outputType>    <defaultAlgorithm>Peter Norvig</defaultAlgorithm>    <difficultyLevel>Medium</difficultyLevel></config>